packages/core/src/browser/frontend-application-module.ts

```diff
@@ -338,4 +339,15 @@ export const frontendApplicationModule = new ContainerModule((bind, unbind, isBo
     bind(ContextMenuContext).toSelf().inSingletonScope();
 
     bind(AuthenticationService).to(AuthenticationServiceImpl).inSingletonScope();
+
+    bind(WindowContribution).toSelf().inSingletonScope();
+    [CommandContribution].forEach(serviceIdentifier =>
+        bind(serviceIdentifier).toService(WindowContribution)
+    );
+    [KeybindingContribution].forEach(serviceIdentifier =>
+        bind(serviceIdentifier).toService(WindowContribution)
+    );
+    [MenuContribution].forEach(serviceIdentifier =>
+        bind(serviceIdentifier).toService(WindowContribution)
+    );
```

Comment:
Can be simplified to:

```suggestion
    bind(WindowContribution).toSelf().inSingletonScope();
    for (const contribution of [CommandContribution, KeybindingContribution, MenuContribution]) {
      bind(contribution).toService(WindowContribution);
    }
```